virtualmachine/vsphere/util.go

```diff
@@ -709,12 +712,12 @@ var reconfigureVM = func(vm *VM, vmMo *mo.VirtualMachine) ([]string, error) {
 		return nil, err
 	}
 
-	dsMo, err := findDatastore(vm, dcMo, vm.datastore)
-	if err != nil {
-		return nil, err
-	}
-
 	for _, disk := range vm.Disks {
+		if disk.Datastore == "" {
```

Comment:
Please add explanatory comments that root-disk datastore will be used by-default.